libraries/AP_NavEKF2/AP_NavEKF2_core.cpp

```diff
@@ -606,6 +608,18 @@ void NavEKF2_core::calcOutputStates()
     // apply a trapezoidal integration to velocities to calculate position
     outputDataNew.position += (outputDataNew.velocity + lastVelocity) * (imuDataNew.delVelDT*0.5f);
 
+    // calculate the average angular rate across the last IMU update
+    // note delAngDT is prevented from being zero in readIMUData()
+    Vector3f angRate = imuDataNew.delAng * (1.0f/imuDataNew.delAngDT);
```

Comment:
That will create sampling interval inconsistencies. The sampling  interval used here is consistent with the sampling interval used to calculate the velocity change from the accel data. In practice noise has not been an issue in the ecl implementation as IMU to body frame origin offsets are usually small and less than 10cm. Notes are provided in the IMU offset parameters about this effect and what to do about it.